src/lib.rs

```diff
@@ -244,8 +244,8 @@ impl Debug for SyncCallback {
 }
 
 impl PartialEq for SyncCallback {
-    fn eq(&self, other: &Self) -> bool {
-        Arc::ptr_eq(&self.0, &other.0)
+    fn eq(&self, _: &Self) -> bool {
+        unimplemented!()
```

Comment:
Why?

Reply:
`#[deny(clippy::vtable_address_comparisons)]`

also, do we really need `fn eq`? If necessary, I can change it to `#[allow]`

Reply:
(see also rust-lang/rust-clippy#6524 about this lint)

Reply:
Technically speaking, this comparison here works as expected, the warning should a false alarm. The comparison tries to feature out whether the two `Arc` are from the same allocation. If and only if they are the same `Arc`, then the comparison succeed. I don't think it's a good idea to leave an panic entry in a lib.

Reply:
Ok i will deal with it

Reply:
done, but ugly

Reply:
Why not just allow the warning?

Reply:
> Why not just allow the warning?

Ok as you wish, done